packages/@ember/routing/route.ts

```diff
@@ -254,14 +248,11 @@ interface Route<T = unknown> extends IRoute<T> {
   error?(error: Error, transition: Transition): boolean | void;
 }
 
-class Route<T = unknown>
-  extends EmberObject.extend(ActionHandler, Evented)
-  implements IRoute, Evented
-{
+class Route<Model = unknown> extends EmberObject.extend(ActionHandler, Evented) implements IRoute {
   static isRouteFactory = true;
 
-  context = {} as T;
-  declare currentModel: T;
+  context = {} as Model;
```

Comment:
It is not. Unfortunately, it is (apparently?) part of the public API of `router.js`'s `Route` type, so we cannot make it private *here* without also making it private *there*, since this is typed as `implements IRoute` (which is the alias for the one for `router.js`). 😩

Reply:
Marked with `@internal` as better-than-nothing signal. b7a7a57c